pytorch.file

```diff
@@ -0,0 +1,138 @@
+## INITENV +PATH PYTHON3PATH %{i}/${PYTHON3_LIB_SITE_PACKAGES}
+
+Source: git+https://github.com/pytorch/pytorch.git?obj=main/v%{realversion}&export=%{n}-%{realversion}&submodules=1&output=/%{n}-%{realversion}.tgz
+Source1: FindEigen3.cmake
+Source99: scram-tools.file/tools/eigen/env
+Patch1: pytorch-missing-braces
+Patch2: pytorch-system-fmt
+Patch3: pytorch-kineto-fmt
+
+BuildRequires: cmake ninja
+Requires: eigen fxdiv numactl protobuf psimd python3 py3-PyYAML py3-pip xz
+Requires: OpenBLAS zlib protobuf fmt py3-pybind11 py3-typing-extensions
+Requires: py3-filelock py3-iniconfig py3-packaging py3-packaging py3-pluggy py3-numpy py3-setuptools
+%if "%{torch_acc}" == "cuda"
+Requires: cuda cudnn
+## INCLUDE cuda-flags
+%elif "%{torch_acc}" == "rocm"
+Requires: rocm aotriton
+## INCLUDE rocm-flags
+%endif
+
+## INCLUDE microarch_flags
+
+%define cuda_arch_float $(echo %{cuda_arch} | tr ' ' '\\n' | sed -E 's|([0-9])$|.\\1|' | tr '\\n' ' ' | sed -e's/ *$/+PTX/')
+
+%prep
+%setup -n %{n}-%{realversion}
+%patch1 -p1
+%patch2 -p1
+%patch3 -p0
+
+%build
+
+%if "%{torch_acc}" == "rocm"
+# Pregenerate some files
+python3 tools/amd_build/build_amd.py
+%endif
+
+cp %{_sourcedir}/FindEigen3.cmake cmake/Modules/
+source %{_sourcedir}/env
+
+# Notice: must be environment variables
+%if "%{torch_acc}" == "cuda"
+export USE_CUDA=ON
+export USE_CUDNN=ON
+export TORCH_CUDA_ARCH_LIST="%{cuda_arch_float}" \
+export nvtx3_dir=${CUDA_ROOT}/include \
+export USE_CUDNN=ON
+export USE_ROCM=OFF
+%elif "%{torch_acc}" == "rocm"
+export USE_CUDA=OFF
+export USE_CUDNN=OFF
+export USE_ROCM=ON
+export ROCM_PATH=${ROCM_ROOT}
+export ROCM_SOURCE_DIR=${ROCM_ROOT}
+export AMDGPU_TARGETS="$(echo '%{rocm_archs}' | sed -e 's/,/ /g')"
+export PYTORCH_ROCM_ARCH="%{rocm_archs}"
+%endif
+
+export MAX_JOBS=%compiling_processes
+export AOTRITON_INSTALLED_PREFIX=${AOTRITON_ROOT}
+export GCC_ROOT=${GCC_ROOT}
+export LibLZMA_ROOT=${XZ_ROOT}
+export USE_SYSTEM_PSIMD=ON
+export PSIMD_SOURCE_DIR=${PSIMD_ROOT}
+export BUILD_CUSTOM_PROTOBUF=OFF
+export USE_SYSTEM_EIGEN_INSTALL=ON
+export USE_SYSTEM_PSIMD=ON
+export USE_SYSTEM_FXDIV=ON
+export USE_SYSTEM_PYBIND11=ON
+export PYBIND11_SYSPATH=${PY3_PYBIND11_ROOT}
+export USE_SYSTEM_BENCHMARK=ON
+
+export BUILD_TEST=OFF
+export BUILD_BINARY=OFF
+export BUILD_PYTHON=ON
+export USE_CUSPARSELT=OFF
+export USE_CUDSS=OFF
+export USE_NCCL=OFF
+export USE_XPU=OFF
+export USE_FBGEMM=OFF
+export USE_KINETO=OFF
+export USE_MAGMA=OFF
+export USE_MPS=OFF
+export USE_NNPACK=OFF
+export USE_QNNPACK=OFF
+export USE_PYTORCH_QNNPACK=OFF
+export USE_XNNPACK=OFF
+export USE_NUMA=ON
+export NUMA_ROOT_DIR=${NUMACTL_ROOT}
+export USE_NUMPY=OFF
+export USE_OPENMP=OFF
```

Comment:
A comment on this:  we tested this change for the threading in Pytorch, and switching OPENMP=OFF did not prevent torch opening multiple threads (probably using some Eigen constructs internally). 

We have solved this problem by using a PyTorchService which disables the threads at once at the beginning of the job. https://github.com/cms-sw/cmssw/pull/47984/files#diff-a27bb569cb776b3eda00ce3807d5e22c68c6d43c1ab000ba4f56cd83b3bd21e5

This option can be switched back on.